fieldsolver/gridGlue.cpp

```diff
@@ -179,6 +189,244 @@ void feedMomentsIntoFsGrid(dccrg::Dccrg<SpatialCell,dccrg::Cartesian_Geometry>&
+          } 
+
+          // Calculate Numerator
+          for (int index = 0; index < halfStencilWidth; index++)
+          {
+            weights.at(2 + index) = 1;
+            weights.at(2 - index) = 1;
+          }
+          // Calculate Denominator
+          int weightDenominator = 2 * halfStencilWidth + 1;
+
+          // Calculate Weights
+          weightL2 = weights.at(0) / weightDenominator;
+          weightL1 = weights.at(1) / weightDenominator;
+          weightC  = weights.at(2) / weightDenominator;
+          weightR1 = weights.at(3) / weightDenominator;
+          weightR2 = weights.at(4) / weightDenominator;
+
+          // Get Arrays
+          Grid   = momentsGrid.get(ii, jj, kk);
+          GridL1 = momentsGrid.get(ii - 1, jj, kk);
+          GridL2 = momentsGrid.get(ii - 2, jj, kk);
+          GridR1 = momentsGrid.get(ii + 1, jj, kk);
+          GridR2 = momentsGrid.get(ii + 2, jj, kk);
+          swap   = swapGrid.get(ii, jj, kk);
+
+          //Blur
+          swap->at(fsgrids::moments::RHOM) = weightC * Grid->at(fsgrids::moments::RHOM) + weightL1 * GridL1->at(fsgrids::moments::RHOM) + weightL2 * GridL2->at(fsgrids::moments::RHOM) + weightR1 * GridR1->at(fsgrids::moments::RHOM) + weightR2 * GridR2->at(fsgrids::moments::RHOM);
+          swap->at(fsgrids::moments::RHOQ) = weightC * Grid->at(fsgrids::moments::RHOQ) + weightL1 * GridL1->at(fsgrids::moments::RHOQ) + weightL2 * GridL2->at(fsgrids::moments::RHOQ) + weightR1 * GridR1->at(fsgrids::moments::RHOQ) + weightR2 * GridR2->at(fsgrids::moments::RHOQ);
+          swap->at(fsgrids::moments::VX) = weightC * Grid->at(fsgrids::moments::VX) + weightL1 * GridL1->at(fsgrids::moments::VX) + weightL2 * GridL2->at(fsgrids::moments::VX) + weightR1 * GridR1->at(fsgrids::moments::VX) + weightR2 * GridR2->at(fsgrids::moments::VX);
+          swap->at(fsgrids::moments::VY) = weightC * Grid->at(fsgrids::moments::VY) + weightL1 * GridL1->at(fsgrids::moments::VY) + weightL2 * GridL2->at(fsgrids::moments::VY) + weightR1 * GridR1->at(fsgrids::moments::VY) + weightR2 * GridR2->at(fsgrids::moments::VY);
+          swap->at(fsgrids::moments::P_11) = weightC * Grid->at(fsgrids::moments::P_11) + weightL1 * GridL1->at(fsgrids::moments::P_11) + weightL2 * GridL2->at(fsgrids::moments::P_11) + weightR1 * GridR1->at(fsgrids::moments::P_11) + weightR2 * GridR2->at(fsgrids::moments::P_11);
+          swap->at(fsgrids::moments::P_22) = weightC * Grid->at(fsgrids::moments::P_22) + weightL1 * GridL1->at(fsgrids::moments::P_22) + weightL2 * GridL2->at(fsgrids::moments::P_22) + weightR1 * GridR1->at(fsgrids::moments::P_22) + weightR2 * GridR2->at(fsgrids::moments::P_22);
+          swap->at(fsgrids::moments::P_33) = weightC * Grid->at(fsgrids::moments::P_33) + weightL1 * GridL1->at(fsgrids::moments::P_33) + weightL2 * GridL2->at(fsgrids::moments::P_33) + weightR1 * GridR1->at(fsgrids::moments::P_33) + weightR2 * GridR2->at(fsgrids::moments::P_33);
+
+          // Reset Weights
+          weights.fill(0);
```

Comment:
This isn't necessary, since they anyway get overwritten in the next loop iteration

Reply:
I think it should be there. If I am right,switching from a 5 point stencil area to a 3 point stencil  area the first and last elements should be zero . The 3 point stencil will overwrite the middle values only though so I will end up with a wrong stencil.

Reply:
Ah, I see. True.!